From 008bf756344a126827f285b7042bf4f2e26fa785 Mon Sep 17 00:00:00 2001
From: Ankit Agarwal <ankit.agarwal@intel.com>
Date: Fri, 2 Aug 2024 21:19:19 +0530
Subject: [PATCH] Updated compatibility and device matrix for some of Hals.

Updated android.hardware.media.c2 hal from hidl V1.0 to aidl V1.
Adding vendor prop for media c2 to enable aidl hal.
Adding media c2 software interface in device compatibility matrix.

Updated hals from hidl to aidl and removed some of hidl from
compatibility matrix-
  android.frameworks.sensorservice
  android.hidl.allocator
  android.hidl.manager
  android.hidl.memory
  android.hidl.token
  android.system.wifi.keystore
  android.hardware.renderscript

Tests: Prepared EB and its booted successfully.

Tracked-On: OAM-122196
Change-Id: I4334a55b6513bd4626c3b53a3a229b06b1decc67
Signed-off-by: Ankit Agarwal <ankit.agarwal@intel.com>
---
 compatibility_matrix.xml | 52 +++++-----------------------------------
 manifest.xml             | 11 +--------
 product.mk               |  5 ++--
 3 files changed, 10 insertions(+), 58 deletions(-)

diff --git a/compatibility_matrix.xml b/compatibility_matrix.xml
index a0bf31f..4424208 100644
--- a/compatibility_matrix.xml
+++ b/compatibility_matrix.xml
@@ -1,53 +1,13 @@
 <compatibility-matrix version="1.0" type="device">
-    <hal format="hidl" optional="false">
+    <hal format="aidl">
         <name>android.frameworks.sensorservice</name>
-        <version>1.0</version>
-        <interface>
-            <name>ISensorManager</name>
-            <instance>default</instance>
-        </interface>
+        <fqname>ISensorManager/default</fqname>
     </hal>
-    <hal format="hidl" optional="false">
-        <name>android.hidl.allocator</name>
-        <version>1.0</version>
-        <interface>
-            <name>IAllocator</name>
-            <instance>ashmem</instance>
-        </interface>
+    <hal format="aidl">
+        <name>android.hardware.media.c2</name>
+        <fqname>IComponentStore/software</fqname>
     </hal>
-    <hal format="hidl" optional="false">
-        <name>android.hidl.manager</name>
-        <version>1.0</version>
-        <interface>
-            <name>IServiceManager</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
-    <hal format="hidl" optional="false">
-        <name>android.hidl.memory</name>
-        <version>1.0</version>
-        <interface>
-            <name>IMapper</name>
-            <instance>ashmem</instance>
-        </interface>
-    </hal>
-    <hal format="hidl" optional="false">
-        <name>android.hidl.token</name>
-        <version>1.0</version>
-        <interface>
-            <name>ITokenManager</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
-    <!--<hal format="hidl" optional="false">
-        <name>android.system.wifi.keystore</name>
-        <version>1.0</version>
-        <interface>
-            <name>IKeystore</name>
-            <instance>default</instance>
-        </interface>
-    </hal>-->
     <!--<vendor-ndk>
-        <version>34</version>
+        <version>35</version>
     </vendor-ndk>-->
 </compatibility-matrix>
diff --git a/manifest.xml b/manifest.xml
index 96f1ba1..cc29553 100644
--- a/manifest.xml
+++ b/manifest.xml
@@ -45,17 +45,8 @@
         <fqname>ICameraProvider/ivi/0</fqname>
     </hal>
 
-    <hal format="hidl">
-        <name>android.hardware.renderscript</name>
-        <transport arch="32+64">passthrough</transport>
-        <version>1.0</version>
-        <interface>
-            <name>IDevice</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <!--<kernel target-level="5"/>-->
     <!--<sepolicy>
-        <version>34.0</version>
+        <version>202504</version>
     </sepolicy>-->
 </manifest>
diff --git a/product.mk b/product.mk
index f40c21a..45c9d4b 100755
--- a/product.mk
+++ b/product.mk
@@ -26,8 +26,7 @@ PRODUCT_PACKAGES +=  \
                     android.hardware.renderscript@1.0-impl \
                     android.hardware.graphics.composer3-service.intel \
                     android.hardware.identity \
-                    android.hardware.identity-service.example \
-                    android.hardware.media.omx@1.0-service
+                    android.hardware.identity-service.example
 
 
 PRODUCT_PROPERTY_OVERRIDES += ro.control_privapp_permissions=enforce
@@ -62,3 +61,5 @@ PRODUCT_COPY_FILES += device/intel/common/host_scripts/start_flash_usb.sh:$(PROD
 PRODUCT_COPY_FILES += vendor/intel/fw/trusty-release-binaries/rpmb_dev:$(PRODUCT_OUT)/scripts/rpmb_dev
 PRODUCT_COPY_FILES += $(LOCAL_PATH)/wakeup.py:$(PRODUCT_OUT)/scripts/wakeup.py
 PRODUCT_COPY_FILES += device/intel/civ/host/vm-manager-binaries/vm-manager.deb:$(PRODUCT_OUT)/scripts/vm-manager.deb
+
+PRODUCT_VENDOR_PROPERTIES += media.c2.hal.selection=aidl
-- 
2.34.1

